runironic-api.sh

```diff
@@ -4,4 +4,12 @@ export IRONIC_DEPLOYMENT="API"
 
 . /bin/configure-ironic.sh
 
-exec /usr/bin/ironic-api --config-file /usr/share/ironic/ironic-dist.conf ${IRONIC_CONFIG_OPTIONS}
+if [[ $IRONIC_API_BEHIND_WSGI == "false" ]]
+then
+  exec /usr/bin/ironic-api --config-file /usr/share/ironic/ironic-dist.conf ${IRONIC_CONFIG_OPTIONS}
+else
+  python3 -c 'import os; import sys; import jinja2; sys.stdout.write(jinja2.Template(sys.stdin.read()).render(env=os.environ))' < /etc/httpd-ironic-api.conf.j2 > /etc/httpd/conf.d/ironic.conf
+  sed -i "/Listen 80/c\#Listen 80" /etc/httpd/conf/httpd.conf
+  exec /usr/sbin/httpd -DFOREGROUND
```

Comment:
There is already an httpd container here, let's not run several more instances of httpd

Reply:
Currently, the PR adds 2 more httpd instance to the picture. If we avoid to add extra httpd instances and make use of the existing one, there will be two problems:
- Web server and WSGI application need to be on the same container.  if we only have one httpd instance, we have to run Ironic-api on the existing httpd container.
- In the PR, a dedicated instance of Apache server acts as a reverse proxy offloading TLS and forward http requests to the conductor. If the existing httpd container also becomes a reverse proxy, that means we are forcing the ironic-api and ironic-conductor to live on the same pod. This may reduce the flexibility of design we can choose for our system.

Reply:
Actually, I think we can split conductor and httpd into different containers, and also make use of the existing httpd instance (used by IPA) to acts as a reverse proxy for the conductor. However, a httpd instance is needed to be in the ironic-api container since we run it as a WSGI application. Therefore, we need at least two httpd instance in the picture, and one instance needs to be in the same container as the ironic-api. WDYT @dtantsur ?

Reply:
First, I'm not convinced that we should use httpd for the conductor. This is a setup nobody has ever used, and I'm not sure what the implications are. I'd leave it alone. I don't see a problem with having ironic-api in the httpd container, after all, this is how non-containerized deployments work. Having 2 httpd containers will raise memory requirements quite severely.

Reply:
@dtantsur  The reason why httpd is used for the conductor is that it also takes care of offloading the encrypted traffic and forward the unencrypted traffic to the conductor. Since we put httpd and the conduct into one K8S pod, I think the unencrypted traffic doesn't hurt too much. The aim is to make Ironic get rid of handling TLS.